src/metpy/calc/tools.py

```diff
@@ -1,1620 +1,1660 @@
+
+    """
+    from pyproj import Geod
+
+    # Inputs must be the same number of dimensions
+    if latitude.ndim != longitude.ndim:
+        raise ValueError('Latitude and longitude must have the same number of dimensions.')
+
+    # If we were given 1D arrays, make a mesh grid
+    if latitude.ndim < 2:
+        longitude, latitude = np.meshgrid(longitude, latitude)
+
+    # pyproj requires ndarrays, not Quantities
+    try:
+        longitude = longitude.m_as('degrees')
+        latitude = latitude.m_as('degrees')
+    except AttributeError:
+        longitude = np.asarray(longitude)
+        latitude = np.asarray(latitude)
+
+    # Determine dimension order for offset slicing
+    take_y = make_take(latitude.ndim, y_dim)
+    take_x = make_take(latitude.ndim, x_dim)
+
+    geod_args = {'ellps': 'sphere'}
+    if kwargs:
+        geod_args = kwargs
+
+    g = Geod(**geod_args)
+
+    forward_az, _, dy = g.inv(longitude[take_y(slice(None, -1))],
+                              latitude[take_y(slice(None, -1))],
+                              longitude[take_y(slice(1, None))],
+                              latitude[take_y(slice(1, None))])
+    dy[(forward_az < -90.) | (forward_az > 90.)] *= -1
+
+    forward_az, _, dx = g.inv(longitude[take_x(slice(None, -1))],
+                              latitude[take_x(slice(None, -1))],
+                              longitude[take_x(slice(1, None))],
+                              latitude[take_x(slice(1, None))])
+    dx[(forward_az < 0.) | (forward_az > 180.)] *= -1
+
+    return dx * units.meter, dy * units.meter
+
+
+@exporter.export
+@preprocess_xarray
+def azimuth_range_to_lat_lon(azimuths, ranges, center_lon, center_lat, **kwargs):
+    """   
+    Convert azimuth and range bins in a polar coordinate system to lat/lon coordinates.
+    Pole refers to the origin of the coordinate system. Adapted from Brian Blaylock's code.
+
+    Parameters
+    - - - - -
+    azimuths :   array_like
+                array of azimuths defining the grid. If not a `pint.Quantity`, assumed to be in degrees.
+    ranges  :    array_like
+                array of range distances from the pole. Typically in meters.
+    center_lat  :  float
+                The latitude of the pole in decimal degrees
+    center_lon :  float
+                The longitude of the pole in decimal degrees
+
+    Returns
+    -------
+    lat, lon:
+        two dimensional arrays of latitudes and longitudes corresponding
+        to the bin positions in the polar coordinate system
+
+    """              
+
+    from pyproj import Geod
+    geod_args = {'ellps': 'sphere'}
+    if kwargs:
+        geod_args = kwargs
+    g = Geod(**geod_args)
+
+    lats = np.ones([len(azimuths),len(ranges)])*center_lat
+    lons = np.ones([len(azimuths),len(ranges)])*center_lon
+    az2D = np.ones_like(lats)*azimuths[:,None]
+    rng2D = np.ones_like(lats)*np.transpose(ranges[:,None])
+    lat,lon,back=g.fwd(lons,lats,az2D,rng2D)

```

Comment:
You can look at the code for `lat_lon_grid_deltas` to see how we handle units there. Since PyProj assumes meters and degrees, the code should use that approach to convert range and azimuth to the correct units. Also, since `back` isn't used, convention is to replace it with `_` (an underscore).

Reply:
I think I used Notepad++ as an editor originally and then Spyder. I manually uploaded to github but assume I could also do this from a command line since I have Git installed on my PC and set up SSH keys accordingly. Yes please, would gladly accept your help in straightening this out.